### PR TITLE
Fix for issue 15735

### LINKED ISSUE #15735: PackageIconUrl is deprecated and should not be recommended
https://docs.microsoft.com/en-us/nuget/reference/nuspec#iconurl:

> iconUrl is deprecated. Use icon instead.

Also, https://docs.microsoft.com/en-us/nuget/reference/msbuild-targets#packing-an-icon-image-file is helpful on how to do this properly.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: fc2b4abb-c382-17ef-f1f3-1413ee27aaa0
* Version Independent ID: 4d4bb4d2-a8f4-f142-7154-40ce7fc7ae32
* Content: [NuGet and .NET libraries](https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/nuget#feedback)
* Content Source: [docs/standard/library-guidance/nuget.md](https://github.com/dotnet/docs/blob/master/docs/standard/library-guidance/nuget.md)
* Product: **dotnet**
* GitHub Login: @JamesNK
* Microsoft Alias: **mairaw**

### PR DESCRIPTION
**Pull request by tonytins**
## Summary

Swapped deprecated ``PackageIconUrl`` for ``PackageIcon`` and ``IconUrl`` to ``Icon``, respectfully, in [docs/standard/library-guidance/nuget.md](https://github.com/dotnet/docs/blob/master/docs/standard/library-guidance/nuget.md)

Fixes #15735

### CODE REVIEW

File: docs/standard/library-guidance/nuget.md [hunk 1]
```diff
@@ -46,7 +46,7 @@ A NuGet package supports many [metadata properties](/nuget/reference/nuspec). Th
 | `Description`                      | `description`              | A long description of the package displayed in UI.             |
 | `Authors`                          | `authors`                  | A comma-separated list of package authors, matching the profile names on nuget.org.             |
 | `PackageTags`                      | `tags`                     | A space-delimited list of tags and keywords that describe the package. Tags are used when searching for packages.             |
-| `PackageIconUrl`                   | `iconUrl`                  | A URL for an image to use as the icon for the package. URL should be HTTPS and the image should be 64x64 and have a transparent background.             |
+| `PackageIcon`                   | `icon`                  | A URL for an image to use as the icon for the package. URL should be HTTPS and the image should be 64x64 and have a transparent background.             |
```
Comment (JamesNK) [inline docs/standard/library-guidance/nuget.md:49]:
```suggestion
| `PackageIcon`                   | `icon`                  | A path to an image in the package to use as a package icon. Read more about [`icon` metadata](/nuget/reference/nuspec#icon). |
```

**Review by JamesNK — APPROVED** [review]
Need to update the description as well.

**Comment by tonytins** [comment]
I went with your suggestion.

**Comment by mairaw** [comment]
Good to merge @JamesNK?

File: docs/standard/library-guidance/nuget.md [hunk 2]
```diff
@@ -46,7 +46,7 @@ A NuGet package supports many [metadata properties](/nuget/reference/nuspec). Th
 | `Description`                      | `description`              | A long description of the package displayed in UI.             |
 | `Authors`                          | `authors`                  | A comma-separated list of package authors, matching the profile names on nuget.org.             |
 | `PackageTags`                      | `tags`                     | A space-delimited list of tags and keywords that describe the package. Tags are used when searching for packages.             |
-| `PackageIconUrl`                   | `iconUrl`                  | A URL for an image to use as the icon for the package. URL should be HTTPS and the image should be 64x64 and have a transparent background.             |
+| `PackageIcon`                   | `icon`                  | A path to an image in the package to use as a package icon. Read more about [`icon` metadata](/nuget/reference/nuspec#icon). |
```
Comment (Youssef1313) [inline docs/standard/library-guidance/nuget.md:49]: @JamesNK, Does `PackageIconFullPath` need to be documented in this article as well? with an explanation whether it's needed together with `PackageIcon` or not?

Comment (Youssef1313) [reply docs/standard/library-guidance/nuget.md:49]: I opened an issue for that so that it doesn't get forgotten if needed to be documented.

**Pull request merged**